### DISCUSSION TITLE
minio mc cp <ERROR> Unable to validate source

### DISCUSSION THREAD

**Discussion by dominikhei** [minio/minio#16794] [Q&A] [answered]
I have a docker-compose project, which creates a Minio Service and an mc service, to create a new bucket and upload a file which is within the minio container in the data/ directory. 

creating the bucket is not problem, but when I want to copy the file in it, I am getting the error: `mc: <ERROR> Unable to validate source myminio/data/yellow_tripdata_2022-01.parquet.`

The complete output from the mc service is: 

    test3-createbuckets-1  | Added `myminio` successfully.
    test3-createbuckets-1  | Bucket created successfully `myminio/sample-lakehouse`.
    test3-createbuckets-1  | mc: <ERROR> Unable to validate source`myminio/data/yellow_tripdata_2022-01.parquet`.
test3-createbuckets-1 exited with code 0

This is my docker-compose.yml file:

    version: '3'
    
    services:
      minio:
        image: minio/minio:latest
        ports:
          - "9000:9000"
          - "9001:9001"
        volumes:
          - ./minio-data:/data
        environment:
          MINIO_ROOT_USER: test
          MINIO_ROOT_PASSWORD: test12334567
        command: server /data --console-address ":9001"
    
      createbuckets:
        image: minio/mc
        depends_on:
          - minio
        entrypoint: >
          /bin/sh -c "
          sleep 60;
          /usr/bin/mc config host add myminio http://minio:9000 test test12334567;
          /usr/bin/mc mb myminio/sample-lakehouse;
          /usr/bin/mc cp myminio/data/yellow_tripdata_2022-01.parquet myminio/sample-lakehouse/yellow_tripdata_2022-01.parquet;
          exit 0;
          "
    
    volumes:
      minio-data:
        driver: local

Does someone have an Idea, why this error occurs? The alias myminio seems to be valid.

**Comment by vadmeste** [comment, marked as answer]
@dominikhei the error means mc is not able to find this object data/yellow_tripdata_2022-01.parquet

What happens if you do `ls ./minio-data/data/yellow_tripdata_2022-01.parquet` ?

**Reply by dominikhei** [reply]
The minio-data folder is local, whilst the data folder in the minio container. In the local minio-data folder is no subfolder with the name data, just the parquet file. 

The parquet file also exists in the containers data directory when execing into it.

**Reply by vadmeste** [reply]
@dominikhei in fact, MinIO needs to have an empty directory when you start it. Users should not access/modify anything inside it.
